addons/block_code/ui/blocks/utilities/template_editor/collapsable_settings.gd

```diff
@@ -0,0 +1,36 @@
+@tool
+class_name CollapsableSettings
+extends HBoxContainer
+
+var collapsed = false
+
+
+# Called when the node enters the scene tree for the first time.
+func _ready() -> void:
+	collapse()
+	$CollapseButton.connect("button_up", collapse)
+	$ExpandButton.connect("button_up", expand)
```

Comment:
For these buttons, please declare the nodes at the top of the script. And also add unique names to them, so the script doesn't depend on UI changes:
```
@onready var _collapse_button: Button = %CollapseButton
@onready var _expand_button: Button = %ExpandButton
```